src/components/dex/Trade.vue

```diff
@@ -357,14 +355,125 @@ export default {
       return this.side === 'Buy' ? this.percentForBuy(value) : this.percentForSell(value);
     },
 
-    percentForBuy() {
-      // TODO: fix this
-      console.log('return');
+    percentForBuy(value) {
+      if (!this.baseHolding || !this.baseHolding.availableBalance
+        || this.baseHolding.availableBalance.isLessThanOrEqualTo(0)) {
+        // TODO: Translate this
+        this.$services.alerts.error(`No balance of ${this.currentMarket.baseCurrency} available to Buy with`);
```

Comment:
shouldn't end a sentence with a preposition; consider `Not enough ${...} balance available to perform a Buy.`